core/src/autogluon/core/trainer/abstract_trainer.py

```diff
@@ -1795,47 +1798,86 @@ def _train_and_save(
-            del model
-        except NoGPUError:
-            logger.warning(f"\tNo GPUs available to train {model.name}... Skipping this model.")
-            self._models_failed_to_train.append(model.name)
-            del model
-        except NotEnoughCudaMemoryError:
-            logger.warning(f"\tNot enough CUDA memory available to train {model.name}... Skipping this model.")
-            self._models_failed_to_train.append(model.name)
-            del model
-        except ImportError as err:
-            logger.error(f"\tWarning: Exception caused {model.name} to fail during training (ImportError)... Skipping this model.")
-            logger.error(f"\t\t{err}")
-            self._models_failed_to_train.append(model.name)
-            if self.verbosity > 2:
-                logger.exception("Detailed Traceback:")
         except Exception as err:
-            logger.error(f"\tWarning: Exception caused {model.name} to fail during training... Skipping this model.")
-            logger.error(f"\t\t{err}")
+            del_model = True
+            if isinstance(err, TimeLimitExceeded):
+                logger.log(20, f"\tTime limit exceeded... Skipping {model.name}.")
+            elif isinstance(err, NotEnoughMemoryError):
+                logger.warning(f"\tNot enough memory to train {model.name}... Skipping this model.")
+            elif isinstance(err, NoValidFeatures):
+                logger.warning(f"\tNo valid features to train {model.name}... Skipping this model.")
+            elif isinstance(err, NoGPUError):
+                logger.warning(f"\tNo GPUs available to train {model.name}... Skipping this model.")
+            elif isinstance(err, NotEnoughCudaMemoryError):
+                logger.warning(f"\tNot enough CUDA memory available to train {model.name}... Skipping this model.")
+            elif isinstance(err, ImportError):
+                logger.error(f"\tWarning: Exception caused {model.name} to fail during training (ImportError)... Skipping this model.")
+                logger.error(f"\t\t{err}")
+                del_model = False
+                if self.verbosity > 2:
+                    logger.exception("Detailed Traceback:")
+            else:  # all other exceptions
+                logger.error(f"\tWarning: Exception caused {model.name} to fail during training... Skipping this model.")
+                logger.error(f"\t\t{err}")
+                if self.verbosity > 0:
+                    logger.exception("Detailed Traceback:")
+            crash_time = time.time()
+            total_time = crash_time - fit_start_time
+            tb = traceback.format_exc()
+            model_info = self.get_model_info(model=model)
             self._models_failed_to_train.append(model.name)
-            if self.verbosity > 0:
-                logger.exception("Detailed Traceback:")
-            del model
+            self._models_failed_to_train_errors[model.name] = dict(
```

Comment:
With the addition of `_models_failed_to_train_errors`, we prolly can get rid of `_models_failed_to_train `

Reply:
good point, removed